dwave/cloud/api/client.py

```diff
@@ -131,9 +133,108 @@ def request(self, method: str, *args, **kwargs):
         return response
 
 
-class VersionedAPISession(LoggingSession):
+class PayloadCompressingSession(LoggingSession):
+    """A :class:`requests.Session` subclass (technically, a further specialized
+    :class:`.LoggingSession`) that adds support for payload compression on the
+    fly.
+
+    Args:
+        compress (bool, default=False):
+            Preemptively compress HTTP request body, even on PUT and POST requests.
```

Comment:
"even on PUT and POST requests" --> to me it's unclear for what other requests the expected behavior is compression that these are now added to